lib/phoenix_html/form.ex

```diff
@@ -631,11 +631,27 @@ defmodule Phoenix.HTML.Form do
   """
   @spec hidden_inputs_for(t) :: list(Phoenix.HTML.safe())
   def hidden_inputs_for(form) do
-    for {k, v} <- form.hidden do
-      hidden_input(form, k, value: v)
+    Enum.flat_map(form.hidden, fn {k, v} ->
+      hidden_inputs_for(form, k, v)
+    end)
+  end
+
+  defp hidden_inputs_for(form, k, values) when is_list(values) do
+    name = input_name(form, k)
+
+    for v <- values do
+      hidden_input(form, k,
+        id: false,
+        name: name <> "[]",
+        value: v
+      )
     end
   end
 
+  defp hidden_inputs_for(form, k, v) do
+    [hidden_input(form, k, value: v)]
```

Comment:
```suggestion
    [hidden_input(form, k, id: false, value: v)]
```

Reply:
Hrm, now I see this may be a breaking change. I guess people could be using this to read a DOM value or something? 🤔 Should we keep the `id: false` only for lists then?

Reply:
Yes, this might break things. Some people like using IDs in tests as well.